pkg/volume/plugins.go

```diff
@@ -123,6 +123,13 @@ type VolumeHost interface {
 	NewWrapperCleaner(spec *Spec, podUID types.UID, mounter mount.Interface) (Cleaner, error)
 }
 
+type RecyclableVolumeHost interface {
+	VolumeHost
+	// RecyclableVolumeConfigFor returns a RecyclableVolumeConfig for a specific volume plugin.
+	// Configs are registered with the host to allow override of the recycling implementation.
+	RecyclableVolumeConfigFor(name string) (*RecyclableVolumeConfig, error)
```

Comment:
Oversight.  Will remove. Originally had the plugins asking the host for its config and the binary could give the host that config (plugin can also provide its own default).  It was suggested to pass config through the plugin constructor instead.